g2p/cli.py

```diff
@@ -531,39 +539,54 @@ def convert(  # noqa: C901
         raise click.UsageError(
             f"Path between '{in_lang}' and '{out_lang}' does not exist"
         )
-    # Figure if the input text is on the command line or in a file
-    input_text_is_a_file = os.path.exists(input_text) and input_text.endswith("txt")
-    if input_text_is_a_file:
-        with open(input_text, encoding="utf8") as f:
-            lines = f.readlines()
-    else:
-        lines = [input_text]
-    # Determine which tokenizer to use, if any
-    if tok is not None and not tok and tok_lang is not None:
-        raise click.UsageError("Specified conflicting --no-tok and --tok-lang options.")
-    if tok is None:
-        tok = True  # Tokenize by default
-    custom_tokenizer = make_tokenizer(tok_lang) if tok_lang else None
-    # Transduce!!!
-    assert in_lang and out_lang
-    transducer = make_g2p(
-        in_lang, out_lang, tokenize=tok, custom_tokenizer=custom_tokenizer
-    )
-    for line in lines:
-        tg = transducer(line)
-        if check:
-            transducer.check(tg, display_warnings=True)
-        outputs = [tg.output_string]
-        if substring_alignments:
-            outputs += [tg.substring_alignments()]
-        if pretty_edges:
-            outputs += [tg.pretty_edges()]
-        if debugger:
-            outputs += [tg.edges, tg.debugger]
-        if len(outputs) > 1:
-            click.echo(pprint.pformat(outputs, indent=4))
+    to_close = None
+    try:
+        if file:
+            if input_text == "-":
+                lines = sys.stdin
+            else:
+                try:
+                    to_close = lines = open(input_text, encoding="utf8")
+                except FileNotFoundError as e:
+                    raise click.UsageError(f"Could not open file {input_text}: {e}")
         else:
-            click.echo(tg.output_string, nl=not input_text_is_a_file)
+            lines = [input_text]
+            if os.path.exists(input_text) and input_text.endswith(".txt"):
+                LOGGER.warning(
+                    "The old heuristic for detecting file input is deprecated, "
+                    f"specify --file if you meant to read your input text from file {input_text}."
+                )
+        # Determine which tokenizer to use, if any
+        if tok is not None and not tok and tok_lang is not None:
+            raise click.UsageError(
+                "Specified conflicting --no-tok and --tok-lang options."
+            )
+        if tok is None:
+            tok = True  # Tokenize by default
+        custom_tokenizer = make_tokenizer(tok_lang) if tok_lang else None
+        # Transduce!!!
+        assert in_lang and out_lang
+        transducer = make_g2p(
+            in_lang, out_lang, tokenize=tok, custom_tokenizer=custom_tokenizer
+        )
+        for line in lines:
+            tg = transducer(line)
+            if check:
+                transducer.check(tg, display_warnings=True)
+            outputs = [tg.output_string]
```

Comment:
assuming this is copy/pasted from the old place, so not checking carefully.

Reply:
Yeah, if you hide whitespace changes you'll see this is old code, just shifted because of the new `try:` block.